.github/workflows/publish.yaml

```diff
@@ -0,0 +1,73 @@
+name: Publish
+on:
+  push:
+    tags:
+      - '*'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      hash: ${{ steps.hash.outputs.hash }}
+    steps:
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      - uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
+        with:
+          python-version: '3.x'
+          cache: pip
+          cache-dependency-path: requirements*/*.txt
+      - run: pip install -r requirements/build.txt
+      # Use the commit date instead of the current date during the build.
+      - run: echo "SOURCE_DATE_EPOCH=$(git log -1 --pretty=%ct)" >> $GITHUB_ENV
+      - run: python -m build
+      # Generate hashes used for provenance.
+      - name: generate hash
+        id: hash
+        run: cd dist && echo "hash=$(sha256sum * | base64 -w0)" >> $GITHUB_OUTPUT
+      - uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+        with:
+          path: ./dist
+  provenance:
+    needs: [build]
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    # Can't pin with hash due to how this workflow works.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v2.0.0
+    with:
+      base64-subjects: ${{ needs.build.outputs.hash }}
+  create-release:
+    # Upload the sdist, wheels, and provenance to a GitHub release. They remain
+    # available as build artifacts for a while as well.
+    needs: [provenance]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
+      - name: create release
+        run: >
+          gh release create --draft --repo ${{ github.repository }}
```

Comment:
It's been a minute since I worked with the `gh release create` command, but I thought `--draft` didn't support using the "auto-generate release notes" button? Anyone know if that's true? Otherwise we'll just have to try it and see... or if someone wants to maintain / enforce a `CHANGES.rst` file, that's fine too.

Reply:
I know we had a conversation about changes in releases vs a file before. I didn't realize GitHub had this limitation. Perhaps the changelog can be generated by editing the release after publishing it as well? It seems like this is more a bug to be fixed by GitHub if it's not possible right now.

Reply:
I remember the details now, and I suspect it's a non-issue. 

The problem arose when I was automating :dependabot: releases to RubyGems... we wanted it entirely automated as soon as someone merged a version bump PR, and the "autogenerate notes" wasn't available via an API call.

But IIRC the button was there... it just required a human.

And since then, they may have updated the API calls anyway, as that was a while ago.

Anyway, we're good for now, and can fix forward if we run into hiccups in the process later.